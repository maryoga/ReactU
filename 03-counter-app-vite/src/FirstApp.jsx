//import { Fragment } from 'react';
//const newMessage = 'MaryoGA';
/* 
const newMessage = {
  message: "Hola mundo",
  title: "MaryoGA",
};
 */
/* const getResult = (a, b) => {
    return a + b;
} */

import PropTypes from 'prop-types'

export const FirstApp = ( { title, subTitle, name } ) => {

  return (
    <>
      <h1 data-testid ="test-title"> { title } </h1>
      {/* <code> { JSON.stringify( newMessage )  } </code> */}
      <p> { subTitle } </p>
      <p> { subTitle } </p>
      <p> { name } </p>
    </>
  );
};

FirstApp.propTypes = {
  name: PropTypes.string,
  subTitle: PropTypes.string,
  title: PropTypes.string.isRequired,
}

FirstApp.defaultProps = {
  name: 'Fernando Herrera',
  subTitle: 'No hay Sub Título',
  // title: 'No hay Título'
}
