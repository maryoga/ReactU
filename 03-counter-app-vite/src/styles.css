html, body {
    background-color: black;
    color:azure;
    font-family: Helvetica, Arial, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}

button {
    padding: 15px;
    font-size: 1.3rem;
}