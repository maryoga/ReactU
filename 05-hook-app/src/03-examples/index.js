
// archivo de barril
export * from './LoadingEpisode';
export * from './MultipleCustomHooks';
export * from './Episode';