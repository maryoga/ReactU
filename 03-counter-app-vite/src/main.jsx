import React from 'react'
import ReactDOM from 'react-dom/client'
import { CounterApp } from './CounterApp';
//import App from './App.jsx'
import './index.css'
//import { FirstApp } from './FirstApp'
//import { HelloWorldApp } from './HelloWorldApp'
import './styles.css';


ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <CounterApp value = { 0 } />
    {/* <FirstApp title="Hola, Soy MaryoGA" /> */}
  </React.StrictMode>,
)
