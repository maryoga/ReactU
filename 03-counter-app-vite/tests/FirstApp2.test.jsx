import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe("Testing on <FirstApp2/>", () => {

  const title = 'Hi, I am Seneca';
  const subTitle = "Sobre la brevedad de la vida";

  test('must match the snapshot', () => {

    const { container} = render( <FirstApp title={ title } />)
    expect( container ).toMatchSnapshot();

  });

  test('It must display the message "Hi, I am Seneca"', () => {
    render( <FirstApp title={ title } />)
    expect( screen.getByText( title ) ).toBeTruthy();
    // screen.debug();
  });

  test('must display the title in an h1', () => {
    render( <FirstApp title={ title } />);
    expect( screen.getByRole( 'heading', { level: 1 }).innerHTML).toContain( title );
  });
  
  test('must show the subtitle sent by props', () => {
    render(
      <FirstApp
        title={ title }
        subTitle = { subTitle }
      />
    );

    expect( screen.getAllByText(subTitle).length ).toBe(2);

  });

});
