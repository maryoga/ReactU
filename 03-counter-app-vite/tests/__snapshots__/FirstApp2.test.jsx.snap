// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on <FirstApp2/> must match the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hi, I am Seneca
     
  </h1>
  <p>
     
    No hay Sub Título
     
  </p>
  <p>
     
    No hay Sub Título
     
  </p>
  <p>
     
    Fernando Herrera
     
  </p>
</div>
`;
